### PR TITLE
Add mem env and sequential file to gen sst in memory

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

**Comment by huachaohuang** [comment]
@siddontang @zhangjinpeng1987 PTAL

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -2866,6 +2867,37 @@ crocksdb_envoptions_t* crocksdb_envoptions_create() {
...
+    return nullptr;
+  }
+  auto file = new crocksdb_sequential_file_t;
+  file->rep = result.release();
+  return file;
+}
+
+size_t crocksdb_sequential_file_read(crocksdb_sequential_file_t* file,
+    size_t n, char* buf, char** errptr) {
+  Slice result;
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/c.cc]: Why result is a Slice?

File: src/rocksdb.rs [hunk 2]
```diff
@@ -1850,6 +1851,97 @@ pub fn supported_compression() -> Vec<DBCompressionType> {
...
+    }
+
+    pub fn skip(&mut self, n: usize) -> Result<(), String> {
+        unsafe {
+            ffi_try!(crocksdb_sequential_file_skip(self.inner, n as size_t));
+            Ok(())
+        }
+    }
+
+    fn ffi_read(&mut self, buf: &mut [u8]) -> Result<usize, String> {
```
Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: Why named ffi_read?

File: librocksdb_sys/crocksdb/c.cc (hunk 1 above)
Comment (zhangjinpeng1987) [reply librocksdb_sys/crocksdb/c.cc]: Please ignore

File: src/rocksdb.rs [hunk 3]
```diff
@@ -1850,6 +1851,97 @@ pub fn supported_compression() -> Vec<DBCompressionType> {
...
+                buf.len() as size_t,
+                buf.as_mut_ptr()
+            ));
+            Ok(size as usize)
+        }
+    }
+}
+
+impl io::Read for SequentialFile {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
```
Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: Seems we don't need ffi_read.

**Comment by zhangjinpeng1987** [comment]
@UncP PTAL

**Comment by UncP** [comment]
LTGM

**Pull request merged**